middleware/secondary: Does not report NOTAUTH when transferring

When using `secondary` to transfer a zone, the server reports an error:

```
[ERROR] Failed to parse transfer `example.com.': dns: no SOA
[ERROR] Failed to transfer example.com.: dns: no SOA
```

The "real" error here is that the `transfer from` server is returning a RCODE of `NOTAUTH`.

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -233,7 +233,7 @@
   branch = "master"
   name = "github.com/miekg/dns"
   packages = ["."]
-  revision = "fb6fbed0f5ec4e418de4f156c18d2e4f9bc854e7"
+  revision = "e78414ef75607394ad7d917824f07f381df2eafa"
 
 [[projects]]
   name = "github.com/opentracing/opentracing-go"

